src/modules/jcms_article_fragment/src/Form/ArticleFragmentForm.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Drupal\jcms_article_fragment\Form;
+
+use Drupal\Core\Entity\ContentEntityForm;
+use Drupal\Core\Entity\EntityManagerInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\jcms_article_fragment\FragmentApi;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Form controller for Article fragment edit forms.
+ *
+ * @ingroup jcms_article_fragment
+ */
+class ArticleFragmentForm extends ContentEntityForm {
+
+  /**
+   * The entity manager.
+   *
+   * @var \Drupal\Core\Entity\EntityManagerInterface
+   */
+  protected $entityManager;
+
+  protected $api;
+
+  /**
+   * Constructs a ContentEntityForm object.
+   *
+   * @param \Drupal\Core\Entity\EntityManagerInterface $entity_manager
+   *   The entity manager.
+   */
+  public function __construct(EntityManagerInterface $entity_manager, FragmentApi $api) {
+    $this->entityManager = $entity_manager;
+    $this->api = $api;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static($container->get('entity.manager'), $container->get('jcms_article_fragment.api'));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    /* @var $entity \Drupal\jcms_article_fragment\Entity\ArticleFragment */
+    $form = parent::buildForm($form, $form_state);
+    if (!$this->entity->isNew()) {
+      $form['new_revision'] = [
+        '#type' => 'checkbox',
+        '#title' => $this->t('Create new revision'),
+        '#default_value' => FALSE,
+        '#weight' => 10,
+      ];
+    }
+    $entity = $this->entity;
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function save(array $form, FormStateInterface $form_state) {
+    // Post image fragment to Lax.
+    try {
```

Comment:
nice, please log the exception message and stack trace somewhere (I guess `error_log` unless there is a more specialized logging service)